### ISSUE TITLE
Exported variable 'X' has or is using name 'Readonly' from external module "typebox" but cannot be named

### ISSUE THREAD

**Issue by TheNoim** [sinclairzx81/typebox#655]
I really hope I don't open yet another issue which is not an issue. But I think I found a bug in the recent release.

Simple repro.: 

Not working: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgFQJ5gKZwL5wGZQQhwDkAAgM7AB2AxgDYCGwUA9DOhgEYQAeJAbgBQQjL0iw4tCNQrwYGOXAC8KTgDoAShgCOAVxYYAJgAo0mdQHkuAKwy0YJhELiu4CuQC41F7YyMy9KiWYDDAMoz0ZhoAyjBQNADmJgCUKULYaUA

Working: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgFQJ5gKZwL5wGZQQhwDkAAgM7AB2AxgDYCGwUA9DOhgEYQAeJAbgBQQjL0iw4tCNQrwYGOXAC8KTgDoAShgCOAVxYYAJgAo0mdQHkuAKwy0YJhELiu4CuQC41F7YyMy9KhmGgDKMFA0AOYmAJSxQtjxQA

It looks like the issue occurs in a combination of `Type.Required` and `Type.ReadonlyOptional`.

This worked in the past without any problems. But I don't know why this happens. It seems like `Readonly` gets exported correctly. And this is an really simple example which should work. 

Thanks in advance.

**Comment by sinclairzx81** [comment]
@TheNoim Hi, thanks for reporting

This is a strange one. I have updated the logic of both TRequired and TPartial and pushed out a fix on 0.31.23. I was able to repro the issue running locally (specifically for TRequired) and that should be resolved on the fix. Currently just waiting on the NPM cache to refresh to test on the TSP. 

Can you test the latest and let me know if this resolves the issue for you?

```
npm install @sinclair/typebox@latest
```
Let me know how you go.

**Comment by TheNoim** [comment]
Yes, tested it. It works as expected again.

**Comment by sinclairzx81** [comment]
@TheNoim All good :)

Will close off this issue for now. Thanks again for reporting.

All the best
S
